crawlers/mooncrawl/mooncrawl/metadata_crawler/cli.py

```diff
@@ -81,167 +86,310 @@ def crawl_uri(metadata_uri: str) -> Any:
+            blockchain_type,
+            {},
+        )
+
+
+        # Process each address
+    for address, tokens in tokens_uri_by_address.items():
+        process_address_metadata_with_leak(
+            address=address,
+            blockchain_type=blockchain_type,
+            batch_size=batch_size,
+            max_recrawl=max_recrawl,
+            threads=threads,
+            tokens=tokens,
+        )
+
+    
+
+
+    spire_jobs = []
+    
+    if spire == True:
+        # Get all jobs for this blockchain from Spire
+        search_query = f"#metadata-job #{blockchain_type.value}"
+        try:
+            entries = get_all_entries_from_search(
+                journal_id=METADATA_TASKS_JOURNAL_ID,
+                search_query=search_query,
+                token=MOONSTREAM_ADMIN_ACCESS_TOKEN,
+                content=True,
+                limit=1000,
+            )
+            
+            logger.info(f"Found {len(entries)} metadata jobs for blockchain {blockchain_type.value}")
+            
+            for entry in entries:
+                try:
+                    if not entry.content:
+                        continue
+                    
+                    job = json.loads(entry.content)
+                    if job.get("blockchain") != blockchain_type.value:
+                        logger.warning(f"Skipping job with mismatched blockchain: {job.get('blockchain')} != {blockchain_type.value}")
+                        continue
+                    spire_jobs.append(job)
+                except Exception as err:
+                    id = entry.entry_url.split("/")[-1]
+                    logger.error(f"Error parsing job from entry {id}: {err}")
+                    continue
+        except Exception as err:
+            logger.error(f"Error fetching jobs from journal: {err}")
+            return
 
-                clean_labels_from_db(
-                    db_session=db_session,
-                    blockchain_type=blockchain_type,
-                    address=address,
-                )
+    # Process each job
+
+    # sessions list for each customer and instance
+    sessions_by_customer: Dict[Tuple[str, str], Session] = {}
+
+    # all sessions in one try block
+    try: 
+        for job in spire_jobs:
+            try:
+                customer_id = job.get("customer_id")
+                instance_id = job.get("instance_id")
+
+                if (customer_id, instance_id) not in sessions_by_customer:
+                    # Create session
+                    # Assume fetch_connection_string fetches the connection string
+                    if custom_db_uri:
+                        connection_string = custom_db_uri
+                    else:
+                        connection_string = request_connection_string(
+                            customer_id=customer_id,
+                            instance_id=instance_id,
+                            token=MOONSTREAM_ADMIN_ACCESS_TOKEN,
```

Comment:
I think moonstream admin do not have access to `seer` credentials, need checks..

Reply:
Moonstream admin do have access to all resources by default i think. But yes we can replace by controller token if it exists.